feat: add multi-day balance calculation to FeeCalculator

yahgw-ai opened the pull request:
## Summary
- Extended FeeCalculator to process all dates in balance data (not just the first)
- Added chronological sorting of dates before processing
- Implemented proper balance change calculation for each day
- Added support for negative balance changes in FileManager validation

## Implementation Details
- First day: balance_change_wei = "0"
- Subsequent days: balance_change_wei = current balance - previous balance
- All dates are now processed and included in the fee report
- Refactored code for better maintainability with extracted helper methods

## Test Plan
✅ Added comprehensive integration tests for multi-day balance calculations
✅ Tests cover:
  - Multiple days with increasing balances
  - Chronological sorting of out-of-order dates
  - Decreasing balances with negative balance changes
  - Mixed balance changes (increases and decreases)
✅ All existing tests updated to reflect new behavior
✅ All tests passing
✅ Lint and type checks passing

Closes #195

yahgw-ai committed:
feat: add multi-day balance calculation to FeeCalculator

- Extended FeeCalculator to process all dates in balance data
- Added chronological sorting of dates before processing
- Implemented balance change calculation for each day
  - First day: balance_change_wei = "0"
  - Subsequent days: balance_change_wei = current - previous balance
- Added support for negative balance changes in FileManager validation
- Updated tests to reflect new multi-day processing behavior

Closes #195

yahgwai commented on src/fee-calculator.ts:
```diff
@@ -26,34 +37,84 @@ export class FeeCalculator {
+    // Create and write fee report
     const feeReport: FeeReport = {
       metadata: {
         chain_id: CHAIN_IDS.ARBITRUM_NOVA,
```
This should always be fetched from the provider, unless it's already present in the file. never use a hardcoded value for chain id

yahgwai commented on src/fee-calculator.ts:
```diff
@@ -26,34 +37,84 @@ export class FeeCalculator {
+    previousBalanceWei: string | null,
+  ): string {
+    if (previousBalanceWei === null) {
+      return FIRST_DAY_BALANCE_CHANGE;
```
Shouldnt the first day balance change just be the balance at the end of that day? we could achieve this by just passing in an initial 0 previousbalance?

yahgw-ai committed:
chore: merge latest changes from develop branch

yahgw-ai committed:
fix: address PR review comments

- Use chain_id from distributors metadata instead of hardcoding ARBITRUM_NOVA
- Change first day balance calculation: now uses current balance - 0 instead of always returning 0
- This makes the first day balance change equal to the balance at end of day
- Updated all tests to reflect new first day behavior

Per reviewer feedback, this provides more accurate representation of the
actual balance change on the first day of tracking.
